src/libaktualizr/primary/secondary_provider.h

```diff
@@ -0,0 +1,26 @@
+#ifndef UPTANE_SECONDARY_PROVIDER_H
+#define UPTANE_SECONDARY_PROVIDER_H
+
+#include <string>
+
+#include "config/config.h"
+#include "storage/invstorage.h"
+#include "uptane/tuf.h"
+
+class SecondaryProvider {
```

Comment:
:+1: I had been considering adding such class too, just needed one more case for "justification" :). Not sure why an instance of this class cannot be passed to Secondary via ctor, though.

Reply:
> I had been considering adding such class too, just needed one more case for "justification"

Yeah, it's the only reasonable way I could find to also add the support for flexible metadata control.

> Not sure why an instance of this class cannot be passed to Secondary via ctor, though.

I was considering the same thing. I might still change it to work that way. I'd originally gone a different route here that more or less required the init() call and I hadn't gotten around to deciding for sure if I wanted to keep it.